src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -283,35 +286,21 @@ private async Task EnsureAnalyzerActionCountsInitializedAsync(AnalyzerDriver dri
         internal async Task<AnalyzerActionCounts> GetOrComputeAnalyzerActionCountsAsync(DiagnosticAnalyzer analyzer, AnalyzerDriver driver, CancellationToken cancellationToken)
         {
             await EnsureAnalyzerActionCountsInitializedAsync(driver, cancellationToken).ConfigureAwait(false);
-            return _lazyAnalyzerActionCountsMap[analyzer];
+            return AnalyzerActionCountsMap[analyzer];
         }
 
         internal AnalyzerActionCounts GetAnalyzerActionCounts(DiagnosticAnalyzer analyzer)
-        {
-            Debug.Assert(_lazyAnalyzerActionCountsMap != null);
-            return _lazyAnalyzerActionCountsMap[analyzer];
-        }
+            => AnalyzerActionCountsMap[analyzer];
 
         private static bool HasActionsForEvent(CompilationEvent compilationEvent, AnalyzerActionCounts actionCounts)
         {
-            if (compilationEvent is CompilationStartedEvent)
-            {
-                return actionCounts.CompilationActionsCount > 0 ||
-                    actionCounts.SyntaxTreeActionsCount > 0 ||
-                    actionCounts.AdditionalFileActionsCount > 0;
-            }
-            else if (compilationEvent is CompilationCompletedEvent)
+            return compilationEvent switch
             {
-                return actionCounts.CompilationEndActionsCount > 0;
-            }
-            else if (compilationEvent is SymbolDeclaredCompilationEvent)
-            {
-                return actionCounts.SymbolActionsCount > 0 || actionCounts.HasAnyExecutableCodeActions;
-            }
-            else
-            {
-                return actionCounts.SemanticModelActionsCount > 0;
-            }
+                CompilationStartedEvent => actionCounts.CompilationActionsCount > 0 || actionCounts.SyntaxTreeActionsCount > 0 || actionCounts.AdditionalFileActionsCount > 0,
+                CompilationCompletedEvent => actionCounts.CompilationEndActionsCount > 0,
+                SymbolDeclaredCompilationEvent => actionCounts.SymbolActionsCount > 0 || actionCounts.HasAnyExecutableCodeActions,
+                _ => actionCounts.SemanticModelActionsCount > 0
+            };
```

Comment:
💭 It doesn't look like this changed anything?

Reply:
I was trying to identify if event can be null or not, and a big if-else chain with pattern matching makes it much harder to reason about null-ness then a smaller, concise switch expression.